### ISSUE TITLE
skaffold builds multiple images (one for each change) 😢

### ISSUE THREAD

**Issue by cheburakshu** [GoogleContainerTools/skaffold#8545]
skaffold build generates multiple images, one for each commit (or with every change), is there a way to prevent this behavior? One time it built images worth 60 GB due to multiple builds/rebuilds. I just want one build (image:latest)

```
⟩ docker images my-service
REPOSITORY     TAG                                                                IMAGE ID       CREATED      SIZE
my-service   ef5fa6514a6c0ebdb16d04063c1da831f3f469647a146ba589108c68b96df693   ef5fa6514a6c   2 days ago   1.39GB
my-service   latest                                                             ef5fa6514a6c   2 days ago   1.39GB
my-service   e77a7046e44df826b930552e3a35c6046a98472e69abdcfc4b05c49902971c32   e77a7046e44d   2 days ago   1.39GB
my-service   cdec8e40e1d2c045a317a710cd3d29a3fa7b4b4f82f2b200264b249e1ef1bb45   cdec8e40e1d2   4 days ago   1.39GB
my-service   33b7444e0f0a077f31db8becb7308d1d609ab5b5b60cd349c0b1d0646a7ccd21   33b7444e0f0a   4 days ago   1.39GB
my-service   6765a190b64805606987ed38ec2bd4946209c3561e0119e04cf792c38d69e8b2   6765a190b648   4 days ago   1.39GB
my-service   d3892392900d518c4699f4af57e2269c02736eba030275b43b72685811eba62b   d3892392900d   4 days ago   1.39GB
my-service   0cc898c05689fbce8e183f6f62ce70622bc10bb78de23f2b9d0ff8b9829f866d   0cc898c05689   4 days ago   1.39GB
my-service   4ac1ffc4241f4148e9a637faba7ee5e3068d20cb4611a20c8445eba222370197   4ac1ffc4241f   4 days ago   1.39GB
my-service   b4b984de04c808110c9de8fed29643a990cc673a75720856cc0de08fe5282a9f   b4b984de04c8   6 days ago   1.38GB
my-service   ef86fb7e0726dd023e8f37d8224b0f88431cac3751b2f5e0392a2f1d2149ba2b   ef86fb7e0726   6 days ago   391MB
my-service   e76caba733c6b21a2b8a6234e748ca573f530e87df6a6f5446db8013355941e4   e76caba733c6   6 days ago   391MB
my-service   e5eb780c7da4535e74fd7473ee533caaffc566725af2bb57e64f78767987f020   e5eb780c7da4   6 days ago   1.75GB
my-service   8f91bd00e818fb90bda625b3b955bbc6106bc42c2b4dcdbecc59318829ba4b35   8f91bd00e818   6 days ago   1.74GB
my-service   3e14e10056d8ec8d6b48ab2825d3c0217a8842a121b6e060843733f88af6c0f4   3e14e10056d8   6 days ago   564MB
my-service   acafc6fbf11d2094263a8ff1e549b723fb9eff937a18bfcc737f911b2375951e   acafc6fbf11d   6 days ago   564MB
my-service   8c1ac934b15e2f26cca92efda4e944c2bb52df0df8b60069602ab3a35cb556e1   8c1ac934b15e   7 days ago   397MB
my-service   f390617d1bb1ce02188af3a9b6f91c1d0183bc879c45a6fcc0eb2a26cb7828d2   f390617d1bb1   7 days ago   397MB
my-service   011007115636110603999a9f98b641a8c74ffad93c23bbf2bf47061a3a6a077a   011007115636   7 days ago   397MB
my-service   58ee2e7338e3c7f979d8f17f8ef3dc7b4151c7ff889cc3ed602bb4a287926bc8   58ee2e7338e3   7 days ago   397MB
```

**Comment by ericzzzzzzz** [comment]
Hi @cheburakshu , thank you for the feedback! I can understand the frustration here. I made this issue as p2 for now, we can adjust that our team have bandwidth to work on it,

**Comment by cheburakshu** [comment]
Sad part of this is, `skaffold` wants to build the image again even when there is no change to the Dockerfile, but just configMap. It is so frustrating when basics are broken.

**Comment by cheburakshu** [comment]
Also, skaffold will build one image for `skaffold run` but wont see that image and will want to build a new one for `skaffold debug`. Anyway, hope these sort of issues will get fixed during the course of time.

**Comment by ericzzzzzzz** [comment]
the requested feature was already implemented #4693, it's just not working as expected, i'll do more investigation to get a fix
